api/internal/git/repospec.go

```diff
@@ -103,75 +101,106 @@ func NewRepoSpecFromURL(n string) (*RepoSpec, error) {
 }
 
 const (
-	refQuery     = "?ref="
-	gitSuffix    = ".git"
-	gitDelimiter = "_git/"
+	refQuery         = "?ref="
+	gitSuffix        = ".git"
+	gitRootDelimiter = "_git/"
 )
 
 // From strings like git@github.com:someOrg/someRepo.git or
 // https://github.com/someOrg/someRepo?ref=someHash, extract
 // the different parts of URL , set into a RepoSpec object and return RepoSpec object.
-func parseGitURL(n string) *RepoSpec {
+func parseGitURL(n string) (*RepoSpec, error) {
 	repoSpec := &RepoSpec{raw: n, Dir: notCloned, Timeout: defaultTimeout, Submodules: defaultSubmodules}
-	// parse query first
-	// safe because according to rfc3986: ? only allowed in query
-	// and not recognized %-encoded
-	beforeQuery, query, _ := strings.Cut(n, "?")
-	n = beforeQuery
-	// if no query, defaults returned
+	// Parse the query first. This is safe because according to rfc3986 "?" is only allowed in
+	// the query and is not recognized %-encoded.
+	// parseQuery returns default values for empty parameters.
+	n, query, _ := strings.Cut(n, "?")
 	repoSpec.Ref, repoSpec.Timeout, repoSpec.Submodules = parseQuery(query)
 
-	if strings.Contains(n, gitDelimiter) {
-		index := strings.Index(n, gitDelimiter)
-		// Adding _git/ to host
```

Comment:
I don't know why it was originally added to the host segment in https://github.com/kubernetes-sigs/kustomize/pull/1271. It's part of the path to to the repo, not the host (our new variable names are nice here!).